rest-service/manager_rest/upload_manager.py

```diff
@@ -1011,3 +1020,128 @@ def _flatten_labels_dict(labels):
     for k, v in labels.items():
         flattened[k] = v.get('values')
     return flattened
+
+
+def _required_bc_changes(yamls):
+    """Find and report backward compatibility issues in the plugin.yaml."""
+    if not yamls:
+        raise RuntimeError("At least one yaml file must be provided")
+
+    if any(filename.endswith('_1_3.yaml') for filename in yamls):
+        return None, None
+
+    # analyse the first yaml file - this is the default behaviour
+    with open(yamls[0]) as fh:
+        try:
+            plugin_yaml = yaml.safe_load(fh)
+        except yaml.YAMLError as e:
+            raise RuntimeError(f"The provided plugin's description "
+                               f"({yamls[0]}) can not be read.\n{e}")
+    incompatible_paths = {'types': [], 'availability_rules': []}
+    for selector in [
+        'data_types::*::properties::*',
+        'inputs::*',
+        'node_types::*::properties::*',
+        'relationships::*::source_interfaces::*::*::inputs::*',
+        'relationships::*::target_interfaces::*::*::inputs::*',
+        'workflows::*::parameters::*',
+    ]:
```

Comment:
I would prefer not to use `::` because eventually we'll again see an issue where someone uses `xxx::` as the name of their node. Perhaps those could just be tuples